src/app/controllers/jugaad.php

```diff
@@ -41,6 +41,14 @@ private function handle_save_file($file) {
                 $template_meta = $this->template_model->get_meta($file["template"]);
                 $orig_data = $this->jugaad_model->get_file_data($file_id, $template_meta, $this->user, false);
 
+                foreach ($template_meta as $name => $metadata) {
+                    if ($metadata['type'] == 'list') {
+                        if ($data[$name] == '') {
+                            $data['name'] = [];
```

Comment:
I did that first, then I realised that this way probably improves the speed. Not that it matters much, but still. Not sure though.